### ISSUE TITLE
Missing support for Atomics or SharedArrayBuffer

### ISSUE THREAD

**Issue by FaFre** [simolus3/sqlite3.dart#174]
I'm unable to run the example with OPFS support.

After clicking on 'Start demo!' i get the following error:
```
Uncaught Unsupported operation: Missing support for Atomics or SharedArrayBuffer! Isolated: false
    at Object.wrapException (http://localhost:8080/web/worker.dart.js:691:17)
    at main (http://localhost:8080/web/worker.dart.js:7780:17)
    at http://localhost:8080/web/worker.dart.js:18022:7
    at http://localhost:8080/web/worker.dart.js:18001:7
    at dartProgram (http://localhost:8080/web/worker.dart.js:18016:5)
    at http://localhost:8080/web/worker.dart.js:18024:3
```
The scripts are served via `dart run tool/example_server.dart`. 

The worker is definitely served with the right headers:
![image](https://github.com/simolus3/sqlite3.dart/assets/59543085/7b3b1a7d-dc82-43ff-abd0-4cd77f9d7219)

I tried with two browsers:
- Mozilla Firefox 116.0
- Chromium 115.0.5790.170 Arch Linux

OS is Manjaro Linux.

**Comment by simolus3** [comment]
Thanks for the report, I see the same behavior and I've fixed the demo server in a60cfbfe40db95fccb1bbc9bfd9424df2c4d896c.

We've only set the header on the worker script itself, not the whole page. I think this used to work, but apparently we need to apply that header to both the worker and the main document now.

**Comment by FaFre** [comment]
Can confirm it is working now! Thanks a lot!
